docker-bake.hcl

```diff
@@ -0,0 +1,72 @@
+// Go version
+variable "GO_VERSION" {
+  default = "1.17"
+}
+// GitHub reference as defined in GitHub Actions (eg. refs/head/master)
+variable "GITHUB_REF" {
+  default = ""
```

Comment:
Looking at this default, so, `docker build --build-arg FOO=""` is not the same as omitting the build-arg (empty string would set it to an empty value, which could potentially override defaults or fall-back behaviour in the Dockerfile); wondering if (like my comment about `GO_VERSION`) it's possible to set build-args conditionally in a bakefile (so "no default, and omit if it's not set")?

Reply:
Ok actually managed to avoid using the `GITHUB_REF` so we only use git metadata for seamless integration.